src/plugins/ui_feature/renderer.ts

```diff
@@ -647,19 +647,30 @@ export class RendererPlugin extends UIPlugin {
           break;
         }
         case "center": {
-          const emptyZone = {
+          const emptyZoneOnTheLeft = {
             ...zone,
-            right: nextColIndex,
             left: previousColIndex,
           };
-          const { x, y, width, height } = this.getters.getVisibleRect(emptyZone);
+          const emptyZoneOnTheRight = {
+            ...zone,
+            right: nextColIndex,
+          };
+          const {
+            x,
+            y,
+            height,
+            width: leftWidth,
+          } = this.getters.getVisibleRect(emptyZoneOnTheLeft);
+          const { width: rightWidth } = this.getters.getVisibleRect(emptyZoneOnTheRight);
+          const halfTextWidth = textWidth / 2;
```

Comment:
the fix looks good. I wonder if we could rely on a single rect to achieve the same result though. Something like 
```javascript
const emptyZone = {
  ...zone,
  right: nextColIndex,
  left: previousColIndex,
};
const { x, y, width, height } = this.getters.getVisibleRect(emptyZone);
const boxMiddle = box.x + box.width / 2;
if (
  x + width < boxMiddle + contentWidth / 2 ||
  x > boxMiddle - contentWidth / 2 ||
  fontSizePX > height
) {
  box.clipRect = { x, y, width, height };
}
```
Which should to the same thing. I find it a bit confusing to work with leftWidth and rightWidth, I had to write it down to know what the code was actually doing. My suggestion (imo , super ubjective, of course) looks more readable? What doyou think ?

Also the names `emptyZoneOnTheRight/Left`  don't seem clear to me :/